tooling/debugger/build.rs

```diff
@@ -70,3 +72,67 @@ fn debug_{test_name}() {{
         .expect("Could not write templated test file.");
     }
 }
+
+fn generate_test_runner_debugger_tests(test_file: &mut File, test_data_dir: &Path) {
+    let test_sub_dir = "noir_test_success";
+    let test_data_dir = test_data_dir.join(test_sub_dir);
+
+    let test_case_dirs = std::fs::read_dir(test_data_dir)
+        .unwrap()
+        .flatten()
+        .filter(|c| c.path().is_dir() && c.path().join("Nargo.toml").exists());
+    let ignored_tests_contents = std::fs::read_to_string("ignored-noir-tests.txt").unwrap();
+    let ignored_tests = ignored_tests_contents.lines().collect::<HashSet<_>>();
+
+    for test_dir in test_case_dirs {
+        let test_file_name =
+            test_dir.file_name().into_string().expect("Directory can't be converted to string");
+        if test_file_name.contains('-') {
+            panic!(
+                "Invalid test directory: {test_file_name}. Cannot include `-`, please convert to `_`"
+            );
+        };
+        let test_dir = &test_dir.path();
+
+        let file_name = test_dir.join("src").join("main.nr");
+        let buf_reader =
+            BufReader::new(File::open(file_name.clone()).expect("Could not open file"));
+        let lines = buf_reader.lines();
+        let test_names: Vec<String> = lines
+            .filter_map(|line_res| {
+                line_res.ok().map(|line| if line.contains("fn test_") { Some(line) } else { None })
+            })
+            .flatten()
+            .collect();
+        for test_name_line in test_names {
+            // TODO: get test name by regex perhaps?
```

Comment:
Optional: I don't know if it's possible, but we could parse these files with the Noir parser, then find all functions that have a `#[test]` attribute on them. Using a visitor should be easy to do, but I don't know if `build.rs` can have another project as a dependency.